api-generation/src/main/scala/ai/h2o/sparkling/api/generation/scala/ParametersTemplate.scala

```diff
@@ -0,0 +1,162 @@
+         |
+         |  //
+         |  // Getters
+         |  //
+         |${generateGetters(parameters)}
+         |
+         |  //
+         |  // Setters
+         |  //
+         |${generateSetters(parameters)}
+         |
+         |  override private[sparkling] def getH2OAlgorithmParams(): Map[String, Any] = {
+         |    super.getH2OAlgorithmParams() ++
+         |      Map(
+         |${generateH2OAssignments(parameters)})
+         |  }
+      """.stripMargin
+    }
+  }
+
+  private def generateParameterDefinitions(parameters: Seq[Parameter]): String = {
+    parameters
+      .map { parameter =>
+        val constructorMethod = resolveParameterConstructorMethod(parameter)
+        s"""  private val ${parameter.swName} = ${constructorMethod}(
+           |    name = "${parameter.swName}",
+           |    doc = "${parameter.comment}")""".stripMargin
+      }
+      .mkString("\n\n")
+  }
+
+  private def generateDefaultValues(parameters: Seq[Parameter], explicitDefaultValues: Map[String, String]): String = {
+    parameters
+      .map { parameter =>
+        val defaultValue = if (parameter.dataType.isEnum) {
+          s"${parameter.dataType.name}.${parameter.defaultValue}.name()"
+        } else {
+          parameter.defaultValue
+        }
+        val finalDefaultValue = explicitDefaultValues.getOrElse(parameter.h2oName, stringify(defaultValue))
+        s"    ${parameter.swName} -> $finalDefaultValue"
+      }
+      .mkString(",\n")
+  }
+
+  private def stringify(value: Any): String = value match {
+    case f: java.lang.Float => s"${f.toString.toLowerCase}f"
+    case d: java.lang.Double => d.toString.toLowerCase
+    case l: java.lang.Long => s"${l}L"
+    case a: Array[_] => s"Array(${a.map(stringify).mkString(", ")})"
+    case v if v == null => null
+    case v => v.toString
+  }
+
+  private def generateGetters(parameters: Seq[Parameter]): String = {
+    parameters
+      .map { parameter =>
+        s"  def get${parameter.swName.capitalize}(): ${resolveParameterType(parameter)} = $$(${parameter.swName})"
+      }
+      .mkString("\n\n")
+  }
+
+  private def generateSetters(parameters: Seq[Parameter]): String = {
+    parameters
+      .map { parameter =>
+        if (parameter.dataType.isEnum) {
+          s"""  def set${parameter.swName.capitalize}(value: String): this.type = {
+             |    val validated = H2OAlgoParamsHelper.getValidatedEnumValue[${parameter.dataType.name}](value)
+             |    set(${parameter.swName}, validated)
+             |  }
+           """.stripMargin
+        } else {
+          s"""  def set${parameter.swName.capitalize}(value: ${resolveParameterType(parameter)}): this.type = {
+             |    set(${parameter.swName}, value)
+             |  }
+           """.stripMargin
+        }
+      }
+      .mkString("\n")
+  }
+
+  private def generateH2OAssignments(parameters: Seq[Parameter]): String = {
```

Comment:
Could we include also the opposite mapping in Algorithms (for the grid) so we can pass SW param names?

Reply:
Np will, do

Reply:
Thanks! The idea is to have mapping (sw name -> h2o name) so we can translate the sw name to internal h2o name in grid